client/js/controllers/editor.js

```diff
@@ -831,6 +831,77 @@
                     }
                 };
 
+             $scope.filesChanged = function(elm){
+                    $scope.files = elm.files
+                    $scope.$apply();
+             }
+
+             $scope.importFile = function () {
+                    var file = filePath;
+                    if (file) {
+                        if (isLeaf) {
+                            var cached = file.split("/");
+                            cached.pop();
+                            file = cached.join("/");
+                        }
+                        var entity = file.split('/');
+                        for(var i=0; i < entity.length; i++){
+                            var test = entity[i] === 'demo';
+                        }
```

Comment:
This is not needed as we have talked over IRC.

You can use the

```
if ($scope.folder !== "demo") 
```

This scope.folder is processed by the editor in the time of something is selected. We could change the name of this var to be more intuitive